includes/class-wcpdf-frontend.php

```diff
@@ -103,27 +104,42 @@ public function woocommerce_api_invoice_number ( $data, $order ) {
 	 * Download invoice frontend shortcode
 	 */
 	public function download_invoice_shortcode( $atts ) {
+		return $this->generate_document_link_shortcode( $atts );
+	}
+
+	/**
+	 * Download pdf frontend general shortcode
+	 */
+	public function download_pdf_shortcode( $atts ) {
+		return $this->generate_document_link_shortcode( $atts );
+	}
+
+	private function generate_document_link_shortcode( $atts ) {
 		global $wp;
 
 		if ( is_admin() ) {
 			return;
 		}
 
 		// Default values
-		$values = shortcode_atts (array(
-			'order_id'		=> '',
-			'link_text'		=> ''
+		$values = shortcode_atts( array(
+			'order_id'      => '',
+			'link_text'     => __(
+				'Download ' . str_replace( '-', ' ', $atts['document_type'] ?? 'invoice' ) . ' (PDF)',
+				'woocommerce-pdf-invoices-packing-slips'
+			),
+			'document_type' => 'invoice'
 		), $atts );
 
-		if ( empty( $values ) ) {
+		if ( ! in_array( $values['document_type'], [ 'invoice', 'packing-slip', 'credit-note', 'proforma' ] ) ) {
```

Comment:
I recommend replacing the explicit array with the function `get_documents()` and a `foreach` loop, because that way you are not limited to those 4 types and custom documents can also take advantage of the shortcode.

```php
$documents = WPO_WCPDF()->documents->get_documents();
foreach ( $documents as $document ) {
	// do you stuff
}
```

Reply:
Thanks for the hint about `get_documents()`, I was thinking about such thing.